v0/destinations/google_ads/transform.js

```diff
@@ -0,0 +1,169 @@
+const get = require("get-value");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG, baseEndpoint } = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  defaultRequestConfig
+} = require("../../util");
+const { EventType } = require("../../../constants");
+
+/**
+ *
+ * @param {Rudder Payload} message
+ * To set app_event_name
+ * The name of any custom app event which is not accepted in the app_event_type field.
+ * This field should contain 1 to 64 Unicode characters (using UTF-8 encoding). This field is required if app_event_type is custom.
+ * https://developers.google.com/app-conversion-tracking/api/request-response-specs#conversion_tracking_request
+ *
+ */
+
+function setEventName(message) {
+  const defaultEventTypes = [
+    "first_open",
+    "session_start",
+    "in_app_purchase",
+    "view_item_list",
+    "view_item",
+    "view_search_results",
+    "add_to_cart",
+    "ecommerce_purchase",
+    "custom"
+  ];
+  const eventName = get(message, "event");
+  /**  check if the event name is one of the default event types.
+   * This field must not contain any of the values reserved for app_event_type.
+   * If a reserved event name is used, the API will return an APP_EVENT_NAME_RESERVED_VALUE error.
+   */
+  const isDefaultName = defaultEventTypes.indexOf(eventName) >= 0;
+  if (isDefaultName) {
+    throw new Error(
+      "Event name is a reserved event name in Google Ads. Aborting Message"
+    );
+  }
+  return eventName;
+}
+/**
+ *
+ * @param {Rudder Payload} message
+ * To set app_event_type
+ * The name of the app event that occurred. This field is an enumeration, and will only accept valued predefined by google ads.
+ * https://developers.google.com/app-conversion-tracking/api/request-response-specs#conversion_tracking_request
+ *
+ */
+function setEventType(message) {
+  const eventName = get(message, "event");
+  switch (eventName.toLowerCase()) {
```

Comment:
Check for `null` or `undefined` before calling `toLowerCase`.

Reply:
we are already checking in the before calling method.. if it is null or undefined it will already throw an error before and not reach this method.